src/widgets/hits/hits.tsx

```diff
@@ -36,65 +51,83 @@ const renderer = ({ renderState, cssClasses, containerNode, templates }) => (
- * @property {string|string[]} [emptyRoot] CSS class to add to the wrapping element when no results.
- * @property {string|string[]} [list] CSS class to add to the list of results.
- * @property {string|string[]} [item] CSS class to add to each result.
- */
-
-/**
- * @typedef {Object} HitsTemplates
- * @property {string|function(object):string} [empty=''] Template to use when there are no results.
- * @property {string|function(object):string} [item=''] Template to use for each result. This template will receive an object containing a single record. The record will have a new property `__hitIndex` for the position of the record in the list of displayed hits.
- */
-
-/**
- * @typedef {Object} HitsWidgetOptions
- * @property {string|HTMLElement} container CSS Selector or HTMLElement to insert the widget.
- * @property {HitsTemplates} [templates] Templates to use for the widget.
- * @property {HitsCSSClasses} [cssClasses] CSS classes to add.
- * @property {boolean} [escapeHTML = true] Escape HTML entities from hits string values.
- * @property {function(object[]):object[]} [transformItems] Function to transform the items passed to the templates.
- */
-
-/**
- * Display the list of results (hits) from the current search.
- *
- * This is a traditional display of the hits. It has to be implemented
- * together with a pagination widget, to let the user browse the results
- * beyond the first page.
- * @type {WidgetFactory}
- * @devNovel Hits
- * @category basic
- * @param {HitsWidgetOptions} $0 Options of the Hits widget.
- * @return {Widget} A new instance of Hits widget.
- * @example
- * search.addWidgets([
- *   instantsearch.widgets.hits({
- *     container: '#hits-container',
- *     templates: {
- *       empty: 'No results',
- *       item: '<strong>Hit {{objectID}}</strong>: {{{_highlightResult.name.value}}}'
- *     },
- *     transformItems: items => items.map(item => item),
- *   })
- * ]);
- */
-export default function hits({
-  container,
-  escapeHTML,
-  transformItems,
-  templates = defaultTemplates,
-  cssClasses: userCssClasses = {},
-}) {
+export type HitsCSSClasses = {
+  /**
+   * CSS class to add to the wrapping element.
+   */
+  root?: string | string[];
+
+  /**
+   * CSS class to add to the wrapping element when no results.
+   */
+  emptyRoot?: string | string[];
+
+  /**
+   * CSS class to add to the list of results.
+   */
+  list?: string | string[];
+
+  /**
+   * CSS class to add to each result.
+   */
+  item?: string | string[];
+};
+
+export type HitsTemplates = {
+  /**
+   * Template to use when there are no results.
+   *
+   * @default 'No Results'
+   */
+  empty?: Template;
+
+  /**
+   * Template to use for each result. This template will receive an object containing a single record. The record will
+   * have a new property `__hitIndex` for the position of the record in the list of displayed hits.
+   *
+   * @default ''
+   */
+  item?: Template<
+    Hit & {
+      __hitIndex: number;
```

Comment:
I think it's not __hitIndex, but position, and thus part of Hit

Reply:
Actually the component used by this widget adds a __hitIndex: https://github.com/algolia/instantsearch.js/blob/ee6a9c657c97adebba9fb9404eae454c3996b86d/src/components/Hits/Hits.js#L34.